TLM/TLM/Manager/Impl/UtilityManager.cs

```diff
@@ -27,23 +26,22 @@ public class UtilityManager : AbstractCustomManager, IUtilityManager {
             var manager = Singleton<VehicleManager>.instance;
 
             for (uint vehicleId = 0; vehicleId < manager.m_vehicles.m_size; ++vehicleId) {
-                ref Vehicle vehicle = ref ((ushort)vehicleId).ToVehicle();
 
-                if (!vehicle.IsValid()) {
+                if (!vehicleId.ToVehicle().IsValid())
                     continue;
-                }
 
-                if ((vehicle.ToExtVehicleType() & filter) == 0) {
+                if ((vehicleId.ToVehicle().ToExtVehicleType() & filter) == 0)
```

Comment:
BTW, have you tested all filters? I doubt we track e.g.: Ferries, Planes, so force despawning may not work. 
I'm asking because you've changed implementation of `ToExtVehicleType()` and now it does not determine vehicle type by its AI.
TM:PE `ExtVehicleType` is tracking only `TrafficManager.Manager.Impl.ExtVehicleManager.VEHICLE_TYPES` everything else is skipped in `_Vehicle.SpawnPatch` or `_VehicleManager.OnCreateVehiclePatch`

The easiest way to check: use debug build and enable Vehicles overlay - if shows `V #ID is a None None` then it's not tracked.

Reply:
Does that also affect your PR #1441 as it's using the same approach (it's not checking vehicle AI in the `MayDespawn` method)?

Reply:
ah, got it, i'll revert some changes - at least this PR will get rid of an unwanted lock :)

Reply:
Ok, reverted to using the vehicle AI to determine type.

I've left the two extensions in that refer to the list in ext vehicle manager in case  you want to use them in #1441